receiver/snmpreceiver/config_test.go

```diff
@@ -0,0 +1,1636 @@
+							{
+								OID: "1",
+								Attributes: []Attribute{
+									{
+										Name:  "a2",
+										Value: "val1",
+									},
+								},
+							},
+						},
+					},
+					"m4": {
+						Unit: "{things}",
+						Sum: &SumMetric{
+							Aggregation: "cumulative",
+							Monotonic:   true,
+							ValueType:   "int",
+						},
+						ScalarOIDs: []ScalarOID{
+							{
+								OID: "1",
+							},
+						},
+					},
+					"m5": {
+						Unit: "{things}",
+						Sum: &SumMetric{
+							Aggregation: "cumulative",
+							Monotonic:   false,
+							ValueType:   "int",
+						},
+						ScalarOIDs: []ScalarOID{
+							{
+								OID: "1",
+								Attributes: []Attribute{
+									{
+										Name:  "a2",
+										Value: "val1",
+									},
+								},
+							},
+						},
+					},
+					"m6": {
+						Unit: "1",
+						Sum: &SumMetric{
+							Aggregation: "delta",
+							Monotonic:   true,
+							ValueType:   "int",
+						},
+						ScalarOIDs: []ScalarOID{
+							{
+								OID: "1",
+								Attributes: []Attribute{
+									{
+										Name:  "a1",
+										Value: "val1",
+									},
+								},
+							},
+							{
+								OID: "2",
+								Attributes: []Attribute{
+									{
+										Name:  "a1",
+										Value: "val2",
+									},
+								},
+							},
+						},
+					},
+					"m7": {
+						Unit: "By",
+						Gauge: &GaugeMetric{
+							ValueType: "int",
+						},
+						ColumnOIDs: []ColumnOID{
+							{
+								OID: "1",
+								ResourceAttributes: []string{
+									"ra1",
+								},
+							},
+						},
+					},
+					"m8": {
+						Unit: "By",
+						Gauge: &GaugeMetric{
+							ValueType: "int",
+						},
+						ColumnOIDs: []ColumnOID{
+							{
+								OID: "1",
+								ResourceAttributes: []string{
+									"ra2",
+								},
+							},
+						},
+					},
+					"m9": {
+						Unit: "By",
+						Gauge: &GaugeMetric{
+							ValueType: "int",
+						},
+						ColumnOIDs: []ColumnOID{
+							{
+								OID: "1",
+								ResourceAttributes: []string{
+									"ra1",
+									"ra2",
+								},
+							},
+						},
+					},
+					"m10": {
+						Unit: "By",
+						Gauge: &GaugeMetric{
+							ValueType: "int",
+						},
+						ColumnOIDs: []ColumnOID{
+							{
+								OID: "1",
+								ResourceAttributes: []string{
+									"ra1",
+									"ra2",
+								},
+								Attributes: []Attribute{
+									{
+										Name:  "a2",
+										Value: "val1",
+									},
+									{
+										Name: "a3",
+									},
+									{
+										Name: "a4",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErr: "",
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			factories, err := componenttest.NopFactories()
+			assert.NoError(t, err)
+
+			factory := NewFactory()
+			factories.Receivers[typeStr] = factory
+			cfg, err := servicetest.LoadConfigAndValidate(filepath.Join("testdata", test.cfgFile), factories)
```

Comment:
Receiver config tests like this have mostly been migrated to use `config.UnmarshalReceiver`. The advantage there is that it runs through the collector's actual decode hooks, so is a more valid test. [Here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/postgresqlreceiver/config_test.go#L120)'s an example, but you can find others throughout the collector.